cmake/Modules/FindFilesystem.cmake

```diff
@@ -0,0 +1,254 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
```

Comment:
IIUC this is from `vector-of-bool/CMakeCM`? Please include provenance here, including the current SHA where this is derived from, and any changes -- so that we don't have to reconstruct in the future.